arc_solvers/models/qa/multi_choice/qa_multi_choice_max_att.py

```diff
@@ -0,0 +1,231 @@
+                                                           "tensor_2_dim": choice_output_dim})
+        self._matrix_attention_question_to_choice = MatrixAttention(
+            SimilarityFunction.from_params(att_question_to_choice_params))
+
+        self._accuracy = CategoricalAccuracy()
+        self._loss = torch.nn.CrossEntropyLoss()
+
+        initializer(self)
+
+    def forward(self,  # type: ignore
+                question: Dict[str, torch.LongTensor],
+                choices_list: Dict[str, torch.LongTensor],
+                label: torch.IntTensor = None,
+                metadata: List[Dict[str, Any]] = None,
+                ) -> Dict[str, torch.Tensor]:
+        # pylint: disable=arguments-differ
+        """
+        Parameters
+        ----------
+        question : Dict[str, torch.LongTensor]
+            From a ``TextField``
+        choices_list : Dict[str, torch.LongTensor]
+            From a ``List[TextField]``
+        label : torch.IntTensor, optional (default = None)
+            From a ``LabelField``
+
+        Returns
+        -------
+        An output dictionary consisting of:
+
+        label_logits : torch.FloatTensor
+            A tensor of shape ``(batch_size, num_labels)`` representing unnormalised log
+            probabilities of the entailment label.
+        label_probs : torch.FloatTensor
+            A tensor of shape ``(batch_size, num_labels)`` representing probabilities of the
+            entailment label.
+        loss : torch.FloatTensor, optional
+            A scalar loss to be optimised.
+        """
+
+        encoded_choices_aggregated = embedd_encode_and_aggregate_list_text_field(choices_list,
+                                                                                 self._text_field_embedder,
+                                                                                 self._embeddings_dropout,
+                                                                                 self._choice_encoder,
+                                                                                 self._choice_aggregate)  # # bs, choices, hs
+
+        encoded_question_aggregated, _ = embedd_encode_and_aggregate_text_field(question, self._text_field_embedder,
+                                                                                self._embeddings_dropout,
+                                                                                self._question_encoder,
+                                                                                self._question_aggregate,
+                                                                                get_last_states=False)  # bs, hs
+
+        q_to_choices_att = self._matrix_attention_question_to_choice(encoded_question_aggregated.unsqueeze(1),
+                                                                     encoded_choices_aggregated).squeeze()
+
+        label_logits = q_to_choices_att
```

Comment:
No need to re-assign variables here.

Reply:
I prefer to keep it like this. It makes it clear that the label logits are the q_to_choices_att.